Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -0,0 +1,396 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Tracing;
+using System.IO;
+using Kudu.Core.Infrastructure;
+using LibGit2Sharp;
+using Kudu.Contracts.Tracing;
+using System.Text.RegularExpressions;
+
+namespace Kudu.Core.SourceControl.Git
+{
+    public class LibGit2SharpRepository : IRepository
+    {
+        private const string _remoteAlias = "external";
+        private readonly ITraceFactory _tracerFactory;
+        private readonly IDeploymentSettingsManager _settings;
+        private readonly GitExeRepository _legacyGitExeRepository;
+
+        public LibGit2SharpRepository(IEnvironment environment, IDeploymentSettingsManager settings, ITraceFactory tracerFactory)
+        {
+            _tracerFactory = tracerFactory;
+            _settings = settings;
+            RepositoryPath = environment.RepositoryPath;
+            _legacyGitExeRepository = new GitExeRepository(environment, settings, tracerFactory);
+        }
+
+        public string CurrentId
+        {
+            get 
+            {
+                using (var repo = new LibGit2Sharp.Repository(RepositoryPath))
```

Comment:
@ahmelsayed @suwatch Actually, the creation of a repository is "quite" cheap. However, depending on the expected load, you may be willing to cache it.

Also, keep in mind that in you alter the index on disk through git.git, the changes *may* not be reflected in LibGit2Sharp  until the next instantiation of the `Repository` (@carlosmn can you confirm that, please?).